als-server/shared/src/main/scala/org/mulesoft/als/server/modules/workspace/resolution/ResolutionTaskManager.scala

```diff
@@ -76,6 +77,13 @@ class ResolutionTaskManager private (
     // TODO: check hot reload to always override editor state profiles
   }
 
+  def addDialectIfNotPresent(state: ALSConfigurationState): Unit = {
+    val newDialect = state.dialects.filterNot(p => repository.getAllFilesUris.contains(p.identifier))
```

Comment:
getAllFilesUris never contains dialects even if are present at the project, right?  is new dialect always equivalent to state.dialects?

Reply:
I think with the update below, the dialects should be in the repository and be shown with `getAllFilesUris`, but could be tested to be sure

Reply:
What update? Maybe I blind, but I cannot find anything related to the getAllFilesUris or getTreeUris methods xd